MrAaronOlsen/trail-up
# Trail create

#### What does  this PR do?
* Refactored user image upload to catch more edge cases and be compatible with trail image loading.
* Refactored user views to call current_user rather than @user
* Added multiple image uploads for trails on trail create.
#### Where should the reviewer start?
* This PR effects Picture model, user and trail model, controller, views: new, edit, show.
#### How should this be manually tested?
* Specs for these have been updated. Run rspec to see lots of Geocoding API Errors, but passing tests.
#### Any background context you want to provide?
* This seems a little hacky at the moment. Loading multiple pictures through nested forms with a polymorphic table adds a few extra steps I'm not sure can be avoided.
* Editing trails has not been added yet.
* Trail show page now needs styling.
#### Questions:
  - Do Migrations Need to be ran?  No
  - Do Environment Variables need to be set? No 
  - Any other deploy steps? No
